src/relax/utils.cc

```diff
@@ -84,5 +84,29 @@ bool IsLeafExpr(const Expr& expr) {
          expr.as<TupleNode>();
 }
 
+class FunctionCopier : public ExprMutator {
+ public:
+  Expr VisitExpr_(const FunctionNode* func) override {
+    // the parameters would be copied and substituted to satisfy the restriction in the well-formed
+    // check: any two functions cannot share the same parameter variable.
+    Array<Var> new_params;
+    for (Var param : func->params) {
+      Var new_param = Var(param->vid, GetStructInfo(param), param->span);
+      this->var_remap_[param->vid] = new_param;
+      new_params.push_back(new_param);
+    }
+
+    Expr body = this->VisitWithNewScope(func->body, new_params);
+
+    return Function(new_params, body, func->ret_struct_info, func->attrs);
+  }
```

Comment:
Consider the case where the Function you want to copy contains another local Function inside. If we override the visit method, we will recurse into the local function, which is not as expected. So let’s change the visit method to a public method, which takes a Function, creates a new set of params, sets the var remapping and recurses into the body.